### PR TITLE
Structure step 2 - silkworm binary

### PR DESCRIPTION
**Pull request by AndreaLanfranchi**
* Given process will open several databases (chaindata, clique, nodes) rename some containers with their context.
* `db::table::create_all` method renamed -> `db::table::deploy_chaindata_tables` with strong check on flags compatibility
* silkworm binary now checks if db is initialized with genesis : if not **and** the chain id is known (we have the genesis file) db is automatically initialized with it
* consistency check amongst CLI arguments `--chain` and/or `--networkid` to match the `chain_config` persisted in db (if any)
* rename `stagedsync::TransactionManager` -> `db::RWTxn`

### CODE REVIEW

File: node/silkworm/db/tables.hpp [hunk 1]
```diff
@@ -139,8 +137,8 @@ inline constexpr db::MapConfig kTables[]{
     kTxLookup,
 };
 
-// Create all tables that do not yet exist.
-void create_all(mdbx::txn& txn);
+//! \brief Ensures all defined tables are present in db with consistent flags. Should a table not exist it gets created
+void deploy_chaindata_tables(mdbx::txn& txn);
```
Comment (yperbasis) [inline node/silkworm/db/tables.hpp:141]: Hmm. Perhaps "create_chaindata_tables"? "deploy" is less clear IMHO.

Comment (AndreaLanfranchi) [reply node/silkworm/db/tables.hpp:141]: Well actually the method does both a check (if table exist it must have compatible flags) or a creation (if table does not exist). I cant find a single word to describe it. Maybe `check_or_create_chaindata_tables` ?

Comment (yperbasis) [reply node/silkworm/db/tables.hpp]: Well, IMHO simply "create" is fine and the checks are just details. For example, we typically call methods just something like "cleanup", even though `cleanup` is probably idempotent and has some checks internally against double cleanup. But I am not insisting.

File: cmd/silkworm.cpp [hunk 2]
```diff
@@ -214,37 +215,85 @@ int main(int argc, char* argv[]) {
...
         chaindata_env.close();
 
     } catch (const CLI::ParseError& ex) {
         return cli.exit(ex);
+    } catch (const std::runtime_error& ex) {
+        log::Error() << ex.what();
+        return -1;
     } catch (const std::invalid_argument& ex) {
-        std::cerr << "Invalid argument :" << ex.what() << "\n" << std::endl;
+        std::cerr << "\tInvalid argument :" << ex.what() << "\n" << std::endl;
```
Comment (yperbasis) [inline cmd/silkworm.cpp:288]: Why not `log::Error` ?

**Review by yperbasis — APPROVED** [review]

File: cmd/silkworm.cpp (hunk 2 above)
Comment (AndreaLanfranchi) [reply cmd/silkworm.cpp:288]: Cause this catch is related to the `try` block where log initialization occurs. We might not have it ready yet.

**Pull request merged**